hnn_core/tests/test_network.py

```diff
@@ -22,7 +22,7 @@ def test_network():
     print(net.cells[:2])
 
     # Assert that proper number of gids are created for Network inputs
-    assert len(net.gid_dict['common']) == 2
+    assert len(net.gid_dict['common']) == 0
```

Comment:
My concern here is that it is nice for user sanity to see that a network object has created gids for `common` inputs without any `common` input spikes occurring when running a simulation. This is consistent with the current convention where e.g. gids are created for `extgauss` and `extpois` regardless of their actual activation in a dipole simulation.

Reply:
It's a bit weird though that you have inputs artificial cells for every imaginable input already in the network. I would think that you want to add input cells on a need basis.

Reply:
+1 for per-need inclusion of inputs into `net`, which the `gid_dict` would then reflect.

Resolving here, but making note in #104 where it seems to fit nicely.